src/meta/src/backup_restore/backup_manager.rs

```diff
@@ -188,7 +188,31 @@ impl<S: MetaStore> BackupManager<S> {
 
     /// Deletes existent backups from backup storage.
     pub async fn delete_backups(&self, ids: &[MetaSnapshotId]) -> MetaResult<()> {
-        self.backup_store.delete(ids).await?;
+        let exclude_id = self
+            .backup_store
+            .manifest()
+            .snapshot_metadata
+            .iter()
+            .map(|s| s.id)
+            .max();
+        let filtered_ids = ids
+            .iter()
+            .filter(|id| {
+                if let Some(exclude_id) = exclude_id && **id == exclude_id{
```

Comment:
I think `delete_backups` is a low-level API and should be allowed to delete backup of specific IDs without concerns. The guarantee of maintaining at least one backup can be maintained externally. What do you think?

Reply:
The latest snapshot is crucial, because its monotonic data (epoch, id) is used to overwrite whenever a snapshot is used.
I'll revert the change to `delete_backups` and find a better place to ensure it later.